### PR TITLE
feat(component): New checkbox component

### PR DESCRIPTION
**Pull request by vasucp1207**
New checkbox component
- Default
- Disabled
- Checked
- Link slot

### CODE REVIEW

**Comment by cogor** [comment]
@vasucp1207  Can you fix typescript errors and resolve conflicts?

**Comment by vasucp1207** [comment]
@cogor, check now.

File: src/components/Checkbox/Checkbox.vue [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<template>
+  <div class="flex gap-3 items-center">
```
Comment (cogor) [inline src/components/Checkbox/Checkbox.vue:2]: In order for a click on a label to change the checkbox status, the entire component needs to be wrapped in a label tag, and the existing one should be replaced with a span.

We could assign an id to the input and label, but that would require generating a unique id. The option of wrapping the entire component in a label tag is simpler.

Full code: 
```vue
<template>
  <label class="flex gap-3 items-center justify-start">
    <input v-model="model" type="checkbox" :disabled="disabled" :class="checkboxClasses" />
    <span v-if="label" :class="labelClasses">{{ label }}</span>
    <slot />
  </label>
</template>
```

**Review by cogor — CHANGES_REQUESTED** [review]

File: src/components/Checkbox/composables/useCheckboxClasses.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  const labelClasses = computed(() => {
+    return defaultLabelClasses
+  })
+
+  return { 
+    checkboxClasses,
+    labelClasses,
+  }
+}
```
Comment (cogor) [inline src/components/Checkbox/composables/useCheckboxClasses.ts:23]:
```suggestion
}

```

**Review by cogor — APPROVED** [review]

**Comment by cogor** [comment]
LGTM

**Pull request merged**